File: app/src/main/java/jp/gr/java_conf/foobar/testmaker/service/SharedPreferenceManager.java
package jp.gr.java_conf.foobar.testmaker.service;

import android.content.Context;
import android.content.SharedPreferences;

public class SharedPreferenceManager {

    private SharedPreferences sharedPreferences;

    public SharedPreferenceManager(Context context) {

        sharedPreferences = context.getSharedPreferences("question", Context.MODE_PRIVATE);

    }

    public int getNumChoose() {
        return sharedPreferences.getInt("num_choose", 3);
    }

    public void setNumChoose(int num) {
        sharedPreferences.edit().putInt("num_choose", num)
                .apply();
    }

    public void setNumWrite(int num) {
        sharedPreferences.edit().putInt("num_write", num)
                .apply();
    }

    public int getNumWrite() {

        return sharedPreferences.getInt("num_write", 1);
    }

    public void setAuto(boolean num) {
        sharedPreferences.edit().putBoolean("auto", num)
                .apply();
    }

    public boolean isAuto() {
        return sharedPreferences.getBoolean("auto", false);
    }

    public void setExplanation(boolean f) {
        sharedPreferences.edit().putBoolean("explanation", f)
                .apply();
    }

    public boolean isExplanation() {
        return sharedPreferences.getBoolean("explanation", false);
    }

    public void setManual(boolean f) {
        sharedPreferences.edit().putBoolean("manual", f)
                .apply();
    }

    public boolean isManual() {
        return sharedPreferences.getBoolean("manual", false);
    }

    public void setRefine(boolean f) {
        sharedPreferences.edit().putBoolean("refine", f)
                .apply();
    }

    public boolean isRefine() {
        return sharedPreferences.getBoolean("refine", false);
    }

    public void setAudio(boolean f) {
        sharedPreferences.edit().putBoolean("audio", f)
                .apply();
    }

    public boolean isAudio() {
        return sharedPreferences.getBoolean("audio", false);
    }

    public void setReverse(boolean f) {
        sharedPreferences.edit().putBoolean("reverse", f)
                .apply();
    }

    public boolean isReverse() {
        return sharedPreferences.getBoolean("reverse", false);
    }


    public void setSort(int i) {
        sharedPreferences.edit().putInt("sort", i).apply();
    }

    public int getSort() {
        return sharedPreferences.getInt("sort", -1);
    }

}

File: app/src/main/java/jp/gr/java_conf/foobar/testmaker/service/SharedPreferenceManager.kt
package jp.gr.java_conf.foobar.testmaker.service

import android.content.Context
import android.content.SharedPreferences

class SharedPreferenceManager(context: Context) {

    private val sharedPreferences: SharedPreferences = context.getSharedPreferences("question", Context.MODE_PRIVATE)

    var numOthers: Int
        get() = sharedPreferences.getInt("num_choose", 3)
        set(num) = sharedPreferences.edit().putInt("num_choose", num)
                .apply()

    var numAnswers: Int
        get() = sharedPreferences.getInt("num_write", 1)
        set(num) = sharedPreferences.edit().putInt("num_write", num)
                .apply()

    var numAnswersSelect: Int
        get() = sharedPreferences.getInt("num_answers_select", 1)
        set(num) = sharedPreferences.edit().putInt("num_answers_select", num)
                .apply()

    var auto: Boolean
        get() = sharedPreferences.getBoolean("auto", false)
        set(num) = sharedPreferences.edit().putBoolean("auto", num)
                .apply()

    var explanation: Boolean
        get() = sharedPreferences.getBoolean("explanation", false)
        set(f) = sharedPreferences.edit().putBoolean("explanation", f)
                .apply()

    var manual: Boolean
        get() = sharedPreferences.getBoolean("manual", false)
        set(f) = sharedPreferences.edit().putBoolean("manual", f)
                .apply()

    var refine: Boolean
        get() = sharedPreferences.getBoolean("refine", false)
        set(f) = sharedPreferences.edit().putBoolean("refine", f)
                .apply()

    var audio: Boolean
        get() = sharedPreferences.getBoolean("audio", false)
        set(f) = sharedPreferences.edit().putBoolean("audio", f)
                .apply()

    var reverse: Boolean
        get() = sharedPreferences.getBoolean("reverse", false)
        set(f) = sharedPreferences.edit().putBoolean("reverse", f)
                .apply()

    var alwaysReview: Boolean
        get() = sharedPreferences.getBoolean("alwaysReview", false)
        set(f) = sharedPreferences.edit().putBoolean("alwaysReview", f)
                .apply()

    var confirmSave: Boolean
        get() = sharedPreferences.getBoolean("confirmSave", false)
        set(f) = sharedPreferences.edit().putBoolean("confirmSave", f)
                .apply()

    var confirmNotes: Boolean
        get() = sharedPreferences.getBoolean("confirmNotes", false)
        set(f) = sharedPreferences.edit().putBoolean("confirmNotes", f)
                .apply()

    var sort: Int
        get() = sharedPreferences.getInt("sort", -1)
        set(i) = sharedPreferences.edit().putInt("sort", i).apply()

    var sortOnline: Int
        get() = sharedPreferences.getInt("sortOnline", 1)
        set(i) = sharedPreferences.edit().putInt("sortOnline", i).apply()


}
